src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -298,142 +305,94 @@ private PlayerID getPlayerID(final Element element, final String attribute, fina
+        "production rule");
   }
 
   /**
    * If mustfind is true and cannot find the productionRule an exception will be thrown.
    */
   private RepairRule getRepairRule(final Element element, final String attribute, final boolean mustFind)
       throws GameParseException {
-    final String name = element.getAttribute(attribute);
-    final RepairRule repairRule = data.getRepairRuleList().getRepairRule(name);
-    if (repairRule == null && mustFind) {
-      throw new GameParseException(mapName, "Could not find production rule. name:" + name);
-    }
-    return repairRule;
+    return getValidatedObject(element, attribute, mustFind, data.getRepairRuleList()::getRepairRule, "repair rule");
   }
 
   /**
    * If mustfind is true and cannot find the territory an exception will be thrown.
    */
   private Territory getTerritory(final Element element, final String attribute, final boolean mustFind)
       throws GameParseException {
-    final String name = element.getAttribute(attribute);
-    final Territory territory = data.getMap().getTerritory(name);
-    if (territory == null && mustFind) {
-      throw new GameParseException(mapName, "Could not find territory. name:" + name);
-    }
-    return territory;
+    return getValidatedObject(element, attribute, mustFind, data.getMap()::getTerritory, "territory");
   }
 
   /**
    * If mustfind is true and cannot find the unitType an exception will be thrown.
    */
   private UnitType getUnitType(final Element element, final String attribute, final boolean mustFind)
       throws GameParseException {
-    final String name = element.getAttribute(attribute);
-    final UnitType type = data.getUnitTypeList().getUnitType(name);
-    if (type == null && mustFind) {
-      throw new GameParseException(mapName, "Could not find unitType. name:" + name);
-    }
-    return type;
+    return getValidatedObject(element, attribute, mustFind, data.getUnitTypeList()::getUnitType, "unitType");
   }
 
   /**
    * If mustfind is true and cannot find the technology an exception will be thrown.
    */
   private TechAdvance getTechnology(final Element element, final String attribute, final boolean mustFind)
       throws GameParseException {
-    final String name = element.getAttribute(attribute);
-    TechAdvance type = data.getTechnologyFrontier().getAdvanceByName(name);
-    if (type == null) {
-      type = data.getTechnologyFrontier().getAdvanceByProperty(name);
-    }
-    if (type == null && mustFind) {
-      throw new GameParseException(mapName, "Could not find technology. name:" + name);
-    }
-    return type;
+    return getValidatedObject(element, attribute, mustFind, name -> {
+      TechAdvance type = data.getTechnologyFrontier().getAdvanceByName(name);
+      if (type == null) {
+        type = data.getTechnologyFrontier().getAdvanceByProperty(name);
+      }
+      return type;
+    }, "technology");
```

Comment:
I think the part I don't like is the function declaration in the middle of listing method parameters. Maybe defining the function first then passing it to the method?